### DISCUSSION TITLE
spdlog-rs's advantage over ftlog?

### DISCUSSION THREAD

**Discussion by ssh352** [SpriteOvO/spdlog-rs#68] [Q&A]
from their benchmark here https://crates.io/crates/ftlog. They are fast too.

Is it possible to add ftlog to spdlog-rs benchmark https://github.com/SpriteOvO/spdlog-rs/blob/main/spdlog/benches/README.md? Thanks!

**Comment by SpriteOvO** [comment]
No problem. Some of the existing benchmarks need to be updated as well, hopefully I can get it done in a week.

**Comment by ssh352** [comment]
cargo clippy runs clean for spdlog-rs. 

lots of lints for ftlog. It appears spdlog-rs code quality is higher.

**Comment by SpriteOvO** [comment]
Just a preview results tested on Windows, I'll be doing some cleanup before pushing updates to the repo.

- `spdlog-rs` (2562c82715fe0b5c51f6fb71a2f74e9bf39e2515)

> [!NOTE]
> We are still preparing `spdlog-rs` v0.4.0 and this result represents that commit only. Performance can probably be further optimized before we finally release v0.4.0.

```
test bench_1_file               ... bench:         182.19 ns/iter (+/- 9.95)
test bench_2_file_async         ... bench:         202.71 ns/iter (+/- 7.91)
test bench_3_rotating_file_size ... bench:         198.77 ns/iter (+/- 24.08)
test bench_4_rotating_daily     ... bench:         182.52 ns/iter (+/- 7.43)
test bench_5_level_off          ... bench:           1.43 ns/iter (+/- 0.03)
```

- `ftlog` (v0.2.14)

```
test bench_1_file               ...                unavailable
test bench_2_file_async         ... bench:         253.61 ns/iter (+/- 15.66)
test bench_3_rotating_file_size ...                unavailable
test bench_4_rotating_daily     ... bench:         250.54 ns/iter (+/- 11.76)
test bench_5_level_off          ... bench:           0.20 ns/iter (+/- 0.00)
```

Benchmark code for `ftlog`: https://github.com/SpriteOvO/spdlog-rs/tree/1309b6a9d523a373970b13c02443fb5f349e1b0d/spdlog/benches/ftlog
